app/script.rb

```diff
@@ -0,0 +1,188 @@
+    browser.goto(ACCOUNTS_URL)
+    puts "Fetching account informatrion"
+    @accounts = Array.new
+    accounts_html.css('div.contracts-section').map do |page|
+      unless page.css('div.section-title.no-data-error').any?
+        name     = page.css('div.main-info').css('a.name').text
+        balance  = page.css('div.primary-balance').css('span.amount').first.text
+        currency = page.css('div.primary-balance').css('span.amount').last.text
+        nature   = page.css('div.section-title.h-small').text.downcase.capitalize
+        account  = Accounts.new(name, balance, currency, nature)
+        @accounts << account
+      end
+    end
+  end
+
+  def check_transactions
+    browser.goto(TRANSACTIONS_URL)
+    set_date
+    sleep 2 
+    puts "Fetching transactions for the last two months"
+    @transactions = Array.new
+    transactions_html.css('li.history-item.success').map do |page|
+      year  = page.xpath('../../preceding-sibling::div[@class = "month-delimiter"]').last.text.split[1]
+      month = page.xpath('../../preceding-sibling::div[@class = "month-delimiter"]').last.text.split[0]
+      day   = page.parent.parent.css('div.day-header').text.split[0]
+      time  = page.css('span.history-item-time').text
+      date  = (year + " " + month + " " + day).to_s
+      description = page.css('span.history-item-description').text.split.join(" ")
+      if !page.css('span.history-item-amount.transaction.income').text.empty?
+        amount = page.css('span.history-item-amount.transaction.income').text
+      elsif !page.css('span.history-item-amount.total').text.empty?
+        amount = page.css('span.history-item-amount.total').text
+      else
+        amount = page.css('span.history-item-amount.transaction').text
+      end
+    transaction = Transactions.new(date, description, amount)
+    @transactions << transaction
+    end
+  end
+
+  def set_date
+    day = Date.today.prev_month(2).day.to_s
+    browser.input(name: 'from').click
+    browser.a(class: %w"ui-datepicker-prev ui-corner-all").click
+    browser.a(text: "#{day}").click
+  end
+
+  def add_transactions
+    @accounts.each do |acount|
+      @transactions.each do |transaction|
+        acount.transactions << transaction
+      end
+    end
+  end
+
+  def assemble
+    hash = Hash.new
+    @accounts.map do |account|
+      account_hash = Hash.new
+      account_hash['name'] = account.name
+      account_hash['balance'] = account.name
+      account_hash['currency'] = account.currency
+      account_hash['nature'] = account.nature
+      account_hash['transactions'] = []
+      account.transactions.map do |transaction|
+        transaction_hash = Hash.new
+        transaction_hash['date'] = transaction.date
+        transaction_hash['description'] = transaction.description
+        transaction_hash['amount'] = transaction.amount
+        account_hash['transactions'].push(transaction_hash)
+      end
+    hash["accounts"] = account_hash
+    hash
+    end
```

Comment:
Something wrong with the indentation. There is no possible way you there can be 2 lines of code one of which is `end` and they are still lined up.